Support project Catalyst

iamDecode opened the pull request:
Using RxSwift in an iOS app compiled for Mac currently yields nasty errors because `UIWebView` is deprecated and not supported in UIKit for Mac:
```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_UIWebView", referenced from:
      objc-class-ref in RxWebViewDelegateProxy.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

The official documentation states `#if !targetEnvironment(UIKitForMac)` is the best way of dealing with these issues ([source](https://developer.apple.com/documentation/uikit/creating_a_mac_version_of_your_ipad_app)).

This PR adds some extra conditions to support compiling with project Catalyst.

freak4pc commented:
This is excellent! If this works in Xcode 10 and builds,  I'm all for it. Let's wait for CI.

freak4pc commented:
So, RxCocoa fails build because of a warning that the targetEnvironment is unknown. 

I'd really want to find away around this, if you have any ideas. 

![image](https://user-images.githubusercontent.com/605076/59793134-7d0e6080-92de-11e9-86b2-78d64e20d906.png)

I thought wrapping it in a `#if compiler(>=5.1)` would work but it doesn't seem like it.

iamDecode commented:
I've stared at this for a while, but I cant seem to find my way around this. It is parsed, even if we do something like `#if false && targetEnvironment(UIKitForMac)` or nesting as @freak4pc suggested. I was not able to silence that specific warning; that does not seem possible for Swift other than silencing all warnings for the target ([relevant source](https://twitter.com/jckarter/status/972894889221414912)).

freak4pc commented:
@iamDecode I opened a bug in Swift Compiler:
https://bugs.swift.org/browse/SR-10973

There seems to be a fix here, we might need to pull this in and bite the bullet, but let me think of options until tomorrow: 
https://github.com/apple/swift/pull/25593

kzaher commented:
I think it should be ok to just add `--allow-warnings` to podfile verification. 

It's probably better to allow users to use it with warnings than have it fail.

freak4pc commented:
Yup. That would be the only option here - it’s a parser but which was fixed but only for Xcode 11. Can you add it to Travis ? @iamDecode

iamDecode commented:
@freak4pc sure thing, I updated it! I added a bit of logic to only add `--allow-warnings` argument for the RxCocoa target, would be a bit much to add it globally.
